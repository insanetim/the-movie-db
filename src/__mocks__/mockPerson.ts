import {
  IExternalIds,
  IPersonCredit,
  IPersonDetails,
} from 'src/interfaces/person.interface'
import { ICredit } from 'src/pages/Credits/types'

const mockExternalIds: IExternalIds = {
  facebook_id: 'facebook_user',
  instagram_id: 'instagram_user',
  tiktok_id: 'tiktok_user',
  twitter_id: 'twitter_user',
}

const mockCreditCast: IPersonCredit[] = [
  {
    character: 'Neo',
    id: 1234,
    popularity: 100500,
    poster_path: '/image',
    release_date: '1999-01-01',
    title: 'The Matrix',
  },
  {
    character: '',
    id: 1223,
    popularity: 100500,
    poster_path: '/image',
    release_date: '2003-01-01',
    title: 'The Matrix Reloaded',
  },
]

const mockedCreditCrew: IPersonCredit[] = [
  {
    department: 'Directing',
    id: 1234,
    job: 'Director',
    popularity: 100500,
    poster_path: '/image',
    release_date: '1999-01-01',
    title: 'The Matrix',
  },
  {
    department: 'Directing',
    id: 4321,
    job: 'Director',
    popularity: 999,
    poster_path: null,
    release_date: '',
    title: 'Jupiter Ascending',
  },
]

const mockedCredits: ICredit[] = [
  {
    key: '1234-the-chosen-one',
    movieSlug: '1234-test-movie',
    posterPath: '/image',
    releaseDate: '1999-01-01',
    releaseDateTitle: '1999',
    role: 'the-chosen-one',
    title: 'the-matrix',
  },
  {
    key: '4321-darth-maul',
    movieSlug: '4321-test-movie',
    posterPath: null,
    releaseDate: '2020-01-01',
    releaseDateTitle: '2020',
    role: 'darth-maul',
    title: 'star-wars',
  },
]

const mockPersonDetails: IPersonDetails = {
  biography: 'test-biography',
  birthday: '1999-01-01',
  external_ids: mockExternalIds,
  gender: 0,
  id: 1234,
  known_for_department: 'Acting',
  movie_credits: {
    cast: mockCreditCast,
    crew: mockedCreditCrew,
  },
  name: 'test-person',
  place_of_birth: 'New York, USA',
  popularity: 100500,
  profile_path: '/image',
}

export { mockPersonDetails, mockedCredits }
