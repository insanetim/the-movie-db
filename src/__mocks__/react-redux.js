export const dispatch = jest.fn()

export const useDispatch = jest.fn(() => dispatch)
export const useSelector = jest.fn(fn => fn())
