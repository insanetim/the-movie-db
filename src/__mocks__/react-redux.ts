const dispatch = jest.fn()
const getState = jest.fn()

const useDispatch = jest.fn(() => dispatch)
const useSelector = jest.fn(fn => fn())

export { dispatch, getState, useDispatch, useSelector }
