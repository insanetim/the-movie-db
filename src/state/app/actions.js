import { v4 as uuid } from 'uuid'

import { NOTIFICATION_DURATION, NOTIFICATION_TYPE } from 'src/constants'

import * as types from './types'

export const showModal = payload => ({
  type: types.SHOW_MODAL,
  payload
})

export const hideModal = () => ({
  type: types.HIDE_MODAL
})

export const showNotification = ({
  id = uuid(),
  messageType = NOTIFICATION_TYPE.SUCCESS,
  messageText,
  duration = NOTIFICATION_DURATION
}) => ({
  type: types.SHOW_NOTIFICATION,
  payload: {
    id,
    messageType,
    messageText,
    duration
  }
})

export const hideNotification = payload => ({
  type: types.HIDE_NOTIFICATION,
  payload
})
