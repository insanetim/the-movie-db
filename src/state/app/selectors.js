import { path } from 'ramda'

export const modalTypeSelector = path(['app', 'modal', 'modalType'])
export const modalPropsSelector = path(['app', 'modal', 'modalProps'])
export const notificationsSelector = path(['app', 'notifications'])
