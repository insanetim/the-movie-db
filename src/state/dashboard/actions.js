import * as types from './types'

export const fetchTrending = page => ({
  type: types.FETCH_TRENDING,
  payload: page
})

export const fetchTrendingRequest = () => ({
  type: types.FETCH_TRENDING_REQUEST
})

export const fetchTrendingSuccess = movies => ({
  type: types.FETCH_TRENDING_SUCCESS,
  payload: movies
})

export const fetchTrendingError = error => ({
  type: types.FETCH_TRENDING_ERROR,
  payload: error
})

export const setTrendingPage = page => ({
  type: types.SET_TRENDING_PAGE,
  payload: page
})

export const fetchSearch = ({ page, query }) => ({
  type: types.FETCH_SEARCH,
  payload: { page, query }
})

export const fetchSearchRequest = () => ({
  type: types.FETCH_SEARCH_REQUEST
})

export const fetchSearchSuccess = movies => ({
  type: types.FETCH_SEARCH_SUCCESS,
  payload: movies
})

export const fetchSearchError = error => ({
  type: types.FETCH_SEARCH_ERROR,
  payload: error
})

export const setSearchPage = page => ({
  type: types.SET_SEARCH_PAGE,
  payload: page
})
