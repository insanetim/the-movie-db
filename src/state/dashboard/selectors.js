import { path } from 'ramda'

export const trendingSelector = path(['dashboard', 'trending'])
export const searchSelector = path(['dashboard', 'search'])
