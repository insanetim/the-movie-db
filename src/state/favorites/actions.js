import * as types from './types'

export const fetchFavorites = page => ({
  type: types.FETCH_FAVORITES,
  payload: page
})

export const fetchFavoritesRequest = page => ({
  type: types.FETCH_FAVORITES_REQUEST,
  payload: page
})

export const fetchFavoritesSuccess = movies => ({
  type: types.FETCH_FAVORITES_SUCCESS,
  payload: movies
})

export const fetchFavoritesFailure = error => ({
  type: types.FETCH_FAVORITES_FAILURE,
  payload: error
})

export const setFavoritesPage = page => ({
  type: types.SET_FAVORITES_PAGE,
  payload: page
})
