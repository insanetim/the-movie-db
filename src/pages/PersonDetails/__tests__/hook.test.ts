import { act, renderHook } from '@testing-library/react'
import { useNavigate, useParams } from 'react-router-dom'
import { mockPersonDetails } from 'src/__mocks__/mockPerson'
import { dispatch } from 'src/__mocks__/react-redux'
import * as personDetailsActions from 'src/store/personDetails/actions'
import * as personDetailsSelectors from 'src/store/personDetails/selectors'

import useContainer from '../hook'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
  useParams: jest.fn(() => ({ personSlug: '1234-john-doe' })),
}))
const navigate = jest.fn()
jest.mocked(useNavigate).mockReturnValue(navigate)

jest.mock('src/store/personDetails/selectors')

describe('PersonDetails useContainer hook', () => {
  jest
    .spyOn(personDetailsSelectors, 'personDetailsLoadingSelector')
    .mockReturnValue(false)
  jest
    .spyOn(personDetailsSelectors, 'personDetailsErrorSelector')
    .mockReturnValue(null)
  const selectPersonById = jest
    .spyOn(personDetailsSelectors, 'personDetailsSelector')
    .mockReturnValue(mockPersonDetails)

  it('should match snapshot', () => {
    const { result } = renderHook(useContainer)

    expect(result.current).toMatchSnapshot()
  })

  it('should match snapshot without params', () => {
    jest.mocked(useParams).mockReturnValueOnce({})
    const { result } = renderHook(useContainer)

    expect(result.current).toMatchSnapshot()
  })

  it('should check "handleGoToCast" method', () => {
    const { result } = renderHook(useContainer)

    act(() => {
      result.current.handleGoToCredits()
    })

    expect(navigate).toHaveBeenCalledWith('credits')
  })

  it('should check "useEffect" method', () => {
    const fetchPersonDetails = jest.spyOn(
      personDetailsActions,
      'fetchPersonDetails'
    )
    selectPersonById.mockReturnValueOnce(undefined as never)
    renderHook(useContainer)

    expect(dispatch).toHaveBeenCalled()
    expect(fetchPersonDetails).toHaveBeenCalledWith(1234)
  })
})
