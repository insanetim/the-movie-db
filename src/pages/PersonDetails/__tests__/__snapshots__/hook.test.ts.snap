// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonDetails useContainer hook should match snapshot 1`] = `
{
  "error": null,
  "loading": false,
  "person": {
    "biography": "test-biography",
    "birthday": "1999-01-01",
    "external_ids": {
      "facebook_id": "facebook_user",
      "instagram_id": "instagram_user",
      "tiktok_id": "tiktok_user",
      "twitter_id": "twitter_user",
    },
    "gender": 0,
    "id": 1234,
    "known_for_department": "Acting",
    "movie_credits": {
      "cast": [
        {
          "character": "Neo",
          "id": 1234,
          "popularity": 100500,
          "poster_path": "/image",
          "title": "The Matrix",
        },
      ],
      "crew": [
        {
          "department": "Directing",
          "id": 1234,
          "job": "Director",
          "popularity": 100500,
          "poster_path": "/image",
          "title": "The Matrix",
        },
        {
          "department": "Directing",
          "id": 4321,
          "job": "Director",
          "popularity": 999,
          "poster_path": "/image",
          "title": "Jupiter Ascending",
        },
      ],
    },
    "name": "test-person",
    "place_of_birth": "New York, USA",
    "popularity": 100500,
    "profile_path": "/image",
  },
}
`;

exports[`PersonDetails useContainer hook should match snapshot without params 1`] = `
{
  "error": null,
  "loading": false,
  "person": {
    "biography": "test-biography",
    "birthday": "1999-01-01",
    "external_ids": {
      "facebook_id": "facebook_user",
      "instagram_id": "instagram_user",
      "tiktok_id": "tiktok_user",
      "twitter_id": "twitter_user",
    },
    "gender": 0,
    "id": 1234,
    "known_for_department": "Acting",
    "movie_credits": {
      "cast": [
        {
          "character": "Neo",
          "id": 1234,
          "popularity": 100500,
          "poster_path": "/image",
          "title": "The Matrix",
        },
      ],
      "crew": [
        {
          "department": "Directing",
          "id": 1234,
          "job": "Director",
          "popularity": 100500,
          "poster_path": "/image",
          "title": "The Matrix",
        },
        {
          "department": "Directing",
          "id": 4321,
          "job": "Director",
          "popularity": 999,
          "poster_path": "/image",
          "title": "Jupiter Ascending",
        },
      ],
    },
    "name": "test-person",
    "place_of_birth": "New York, USA",
    "popularity": 100500,
    "profile_path": "/image",
  },
}
`;
